### PR TITLE
CLIO - 40109 - Add option to provide a string to thModal as its template

### PR DESCRIPTION
**Pull request by frankieyan**

### CODE REVIEW

File: themis_components/thModal/thModalManager.service.coffee [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
     modals = []
 
-    show = ({path, name, params, context} = {}) ->
-      path ?= ""; name ?= path; params ?= ""; context ?= {} # set defaults
+    show = ({path, name, params, context, template} = {}) ->
+      # set defaults
+      path ?= ""
+      name ?= path
+      params ?= ""
+      context ?= {}
```
Comment (JefCurtis) [inline themis_components/thModal/thModalManager.service.coffee]: Should we be setting a default value for `template` here as well?

File: themis_components/thModal/thModalManager.service.coffee [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
 
       deferred = $q.defer()
-      modalPromise = $http(url: path, method: "GET", params: params)
+      modalPromise = if template?
+        $q.when({data: template})
+      else
+        $http(url: path, method: "GET", params: params)
+
```
Comment (JefCurtis) [inline themis_components/thModal/thModalManager.service.coffee]: Feel like this reads a little better. Just for indentation reasons? `#style-guide`
```coffeescript
modalPromise =
  if template? 
    $q.when({data: template})
  else
    $http(url: path, method: "GET", params: params)
```

File: themis_components/thModal/thModalManager.service.spec.coffee [hunk 3]
```diff
@@ -35,36 +36,46 @@ describe 'ThemisComponents: Service: thModalManager', ->
...
-        performAction("invalidTemplate.html", 'invalid')
+      it "rejects template silently", ->
+        performAction("invalidTemplate.html", "invalid")
         expect(ModalManager._modals.length).toBe 0
 
-  describe '#dismiss()', ->
+    context "with inline template", ->
+      inlineTemplate = "<h1>Test</h1>"
+
+      beforeEach ->
```
Comment (JefCurtis) [inline themis_components/thModal/thModalManager.service.spec.coffee]: This probably doesn't need a `beforeEach` block as you are really only applying it to one test.

File: themis_components/thModal/thModalManager.service.spec.coffee [hunk 4]
```diff
@@ -1,13 +1,14 @@
 context = describe
-describe 'ThemisComponents: Service: thModalManager', ->
-  ModalManager = httpBackend = promise = null
+describe "ThemisComponents: Service: thModalManager", ->
+  ModalManager = httpBackend = promise = $scope = null
 
   beforeEach ->
-    angular.mock.module 'ThemisComponents'
+    angular.mock.module "ThemisComponents"
```
Comment (JefCurtis) [inline themis_components/thModal/thModalManager.service.spec.coffee]: Double quotes: 👍 👍 👍 👍 👍 👍

File: themis_components/thModal/examples/5 - Inline content/coffee.coffee [hunk 5]
```diff
@@ -0,0 +1,24 @@
+angular.module("thDemo", ["ThemisComponents"])
+  .controller "DemoCtrl", (ModalManager) ->
+    @headerString = "thModal Example 5"
```
Comment (JefCurtis) [inline themis_components/thModal/examples/5 - Inline content/coffee.coffee]: Does this need to be be on `this` here? Seems like you aren't using it in the view anyways?

File: themis_components/thModal/thModalManager.service.coffee (hunk 1 above)
Comment (frankieyan) [inline themis_components/thModal/thModalManager.service.coffee]: I was gonna initialize it to an empty string - but that wasn't going to pass a template? test. What's the best practice here? Initialize to null?

File: themis_components/thModal/thModalManager.service.coffee (hunk 2 above)
Comment (frankieyan) [inline themis_components/thModal/thModalManager.service.coffee]: Will change - summoning @slapheadted

File: themis_components/thModal/examples/5 - Inline content/coffee.coffee (hunk 5 above)
Comment (frankieyan) [inline themis_components/thModal/examples/5 - Inline content/coffee.coffee]: Fixed in ced5a18

File: themis_components/thModal/thModalManager.service.spec.coffee (hunk 3 above)
Comment (frankieyan) [inline themis_components/thModal/thModalManager.service.spec.coffee]: Fixed in ced5a18

File: themis_components/thModal/thModalManager.service.coffee (hunk 1 above)
Comment (JefCurtis) [inline themis_components/thModal/thModalManager.service.coffee]: My first instinct would be to write it like this. 
```coffeescript
show = (options = {}) ->
  {
    path = ""
    name = path
    params = ""
    context = {}
    template = null
  } = options
```
So ya maybe I would initialize to `null`. What do you think?

Comment (frankieyan) [inline themis_components/thModal/thModalManager.service.coffee]: Looks much better 👍  Copied!

File: themis_components/thModal/thModalManager.service.coffee [hunk 6]
```diff
@@ -1,12 +1,23 @@
-angular.module('ThemisComponents')
-  .factory 'ModalManager', ($http, $q) ->
+angular.module("ThemisComponents")
+  .factory "ModalManager", ($http, $q) ->
```
Comment (seanhealy) [inline themis_components/thModal/thModalManager.service.coffee]: Very nice!

**Pull request merged**